lib/Frontend/Frontend.cpp

```diff
@@ -631,6 +633,15 @@ Optional<unsigned> CompilerInstance::getRecordedBufferID(const InputFile &input,
   }
   auto buffers = getInputBuffersIfPresent(input);
 
+  // For non-primary '.swift' files, recover by dummy buffer so that the primary
+  // files are diagnosed. Also, IDE functionalities want to proceed even with
+  // missing files.
+  if (!buffers.hasValue() && input.getType() == file_types::TY_Swift &&
```

Comment:
Is there a way to restrict this to IDE input modes? I'm hesitant to have compilation proceed in general if the user has managed to nuke an input.

Reply:
I thought it was a good thing to proceed to typechecking to get further diagnostics. Since we've already emitted diagnostics for the missing files, it won't go beyond typechecking anyway.

But sure, I can do it only for IDE requests.

Reply:
@CodaFi Added a commit to make the recovery "opt-in" and enable it only in SourceKit. How does it look?